### PR TITLE
Developer guide

### PR DESCRIPTION
**Pull request by LefterisJP**
Please provide feedback if this accurately reflects what we want to have as a developer's/contributor's guide in Raiden.

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,147 @@
+# Raiden Development Guide
+
+Welcome! This guide serves as the guideline to contributing to the Raiden Network
+codebase. It's here to help you understand what developmenr practises we use here
```
Comment (czepluch) [inline CONTRIBUTING.md]: typo. s/developmenr/development

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,147 @@
...
+    - [Creating an Issue](#creating-an-issue)
+    - [Creating a Pull Request](#creating-a-pull-request)
+- [Development Guidelines](#development-guidelines)
+    - [Coding Style](#coding-style)
+    - [Workflow](#workflow)
+
+
+## Contributing
+
+You can contribute to the development with two basic methods. You can either open
```
Comment (czepluch) [inline CONTRIBUTING.md]: Maybe reformulate to: There are two ways you can contribute to the development.

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -0,0 +1,147 @@
...
+- What you were expecting to happen instead
+
+### Creating a Pull Request
+
+If you have some coding abilities and would like to contribute to the actual codebase of Raiden then you can open a Pull Request(PR) against the repository.
+
+All PRs should be:
+- Self-contained
+- As short as possible and address a single issue or even a part of an issue.
+If the PR adds a new feature and it's very long, consider breaking into smaller ones.
```
Comment (czepluch) [inline CONTRIBUTING.md]: consider breaking _it_ into smaller ones

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -0,0 +1,147 @@
...
+
+In order for a Pull Request to get merged into the main repository you should have one
+approved review from one of the core developers of Raiden and also all Continuous integration tests should be passing and the CI build should be green.
```
Comment (czepluch) [inline CONTRIBUTING.md]: Either capitalize both or neither in "Continuous integration"

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -0,0 +1,147 @@
...
+
+
+## Contributing
+
+You can contribute to the development with two basic methods. You can either open
+an Issue or if you have programming abilities open a Pull Request.
+
+### Creating an Issue
+
+If you experience a problem while using Raiden or want to request a feature then you should open an issue against the repository. All issues should contain:
```
Comment (utzig) [inline CONTRIBUTING.md]: No need for `against the repository` when talking about issues I think.

File: CONTRIBUTING.md (hunk 3 above)
Comment (utzig) [inline CONTRIBUTING.md]: Could be worded as something like `consider breaking very long PRs into smaller ones` independent of being a new feature or not.

**Comment by utzig** [comment]
@czepluch I never really wrote solidity code, but would be cool adding a section about the `flake8` like linter and rules to follow for solidity.

File: CONTRIBUTING.md [hunk 6]
```diff
@@ -0,0 +1,200 @@
...
+```
+
+#### Solidity
+
+For solidity we generally follow the style guide as shown in the [solidity documentation](http://solidity.readthedocs.io/en/develop/style-guide.html)
+with a few notable exceptions:
+
+** Variable Names **
+
+All variable name should be in snake case, just like in python. Function names on the other hand stay in CamelCase.
```
Comment (czepluch) [inline CONTRIBUTING.md]: Function names are mixedCase, no? They start with a lower case letter.

Comment (LefterisJP) [inline CONTRIBUTING.md]: Good point. Let me fix.

Comment (czepluch) [inline CONTRIBUTING.md]: Okay. So camel case can be both. mixedCase seems to be a python term, that is the definition of what we want.

Comment (czepluch) [inline CONTRIBUTING.md]: https://en.wikipedia.org/wiki/Camel_case#Variations_and_synonyms

**Comment by LefterisJP** [comment]
@czepluch Rebased on latest master and also integrated the suggested change. Please approve as soon as you can.

**Pull request merged**